homeassistant/components/sensor/mysensors.py

```diff
@@ -0,0 +1,108 @@
+"""
+homeassistant.components.sensor.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MySensors sensors.
+
+Config:
+  sensor:
+    - platform: mysensors
+      port: '/dev/ttyACM0'
+"""
+import logging
+
+# pylint: disable=no-name-in-module, import-error
+import homeassistant.external.pymysensors.mysensors.mysensors as mysensors
+import homeassistant.external.pymysensors.mysensors.const as const
+from homeassistant.helpers.entity import Entity
+
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL, EVENT_HOMEASSISTANT_STOP, TEMP_CELCIUS)
+
+CONF_PORT = "port"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Setup the mysensors platform. """
+
+    devices = {}    # keep track of devices added to HA
+
+    def sensor_update(update_type, nid):
+        """ Callback for sensor updates from the MySensors gateway. """
+        sensor = gateway.sensors[nid]
+        if sensor.sketch_name is None:
+            return
+        if nid not in devices:
+            devices[nid] = MySensorsNode(sensor.sketch_name)
+            add_devices([devices[nid]])
+
+        node = devices[nid]
+        node.battery_level = sensor.battery_level
+        for child_id, child in sensor.children.items():
+            node.update_child(child_id, child)
+        node.update_ha_state()
+
+    port = config.get(CONF_PORT)
+    if port is None:
+        _LOGGER.error("Missing required key 'port'")
+        return False
+
+    gateway = mysensors.SerialGateway(port, sensor_update)
+    gateway.start()
+
+    # Just assume celcius means that the user wants metric for now.
+    # It may make more sense to make this a global config option in the future.
+    gateway.metric = (hass.config.temperature_unit == TEMP_CELCIUS)
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                         lambda event: gateway.stop())
+
+
+class MySensorsNode(Entity):
+    """ Represents a MySensors node. """
+    def __init__(self, name):
+        self._name = name
+        self.battery_level = 0
+        self.children = {}
+
+    @property
+    def should_poll(self):
+        """ MySensor gateway pushes its state to HA.  """
+        return False
+
+    @property
+    def name(self):
+        """ The name of this sensor. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return ''
```

Comment:
Yeah.  There are many different sensor types and the state could probably be dependent on the type (i.e. use STATE_ON/OFF for switches, STATE_OPEN/CLOSED for doors, etc.).  However, the problem comes when a node has multiple child sensors attached with differing states.